yarn-project/types/src/interfaces/aztec_rpc.ts

```diff
@@ -158,6 +159,17 @@ export interface AztecRPC {
    */
   getTxReceipt(txHash: TxHash): Promise<TxReceipt>;
 
+  /**
+   * Retrieves the private storage data at a specified contract address and storage slot.
+   * The returned data is data at the storage slot or throws an error if the contract is not deployed.
+   *
+   * @param owner - The address for whom the private data is encrypted.
+   * @param contract - The AztecAddress of the target contract.
+   * @param storageSlot - The Fr representing the storage slot to be fetched.
+   * @returns A set of note preimages for the owner in that contract and slot.
+   */
+  getPrivateStorageAt(owner: AztecAddress, contract: AztecAddress, storageSlot: Fr): Promise<NotePreimage[]>;
```

Comment:
It all looks fine, just boils down to your thoughts @iAmMichaelConnor on re-introducing this method on the RPC server. We used to have it, then we removed it as you expressed concerns about making private state freely accessible. There is a feeling that being able to read private state directly aids in the dev/testing process.

Reply:
Yup, when reading old conversations I saw that that removal came out of a suggestion of mine. But I changed my mind after going through this testing guide.